torch/_dynamo/eval_frame.py

```diff
@@ -1531,6 +1531,13 @@ def patch():
             # disable future hooking
             opt.step.hooked = True  # type: ignore[attr-defined]
 
+        # annoying - we cannot annotate these functions directly due to
+        # torch.package issues, see https://github.com/pytorch/pytorch/pull/116229
+        if config.trace_distributed and torch.distributed.is_available():
+            import torch.distributed.fsdp._flat_param as flat_param
+
+            torch._dynamo.allow_in_graph(flat_param._same_storage_size)
```

Comment:
This is an internal util of FSDP, so I kinda loathe testing this? Anyway, the e2e FSDP tests will cover this.

Reply:
cc @bdhirsh 

It's not at all obvious to me that `make_fx` will be able to accurately trace through this? Because we don't have any storage tracing at all.

Reply:
So yeah, I kind of do want a test! Are we installing sufficient guards, what is the output graph after aot autograd.

Reply:
it works in the e2e FSDP test just fine. Happy to add a test, but I don't think that's the right thing to do with a private API.

Reply:
re make_fx/aot_autograd

```

TRACED GRAPH
 ===== Forward graph 1 =====
 <eval_with_key>.7 from /data/users/voz/pytorch/torch/fx/experimental/proxy_tensor.py:637 in wrapped class <lambda>(torch.nn.Module):
    def forward(self, arg0_1: "f32[2, 2]"):
        # File: /data/users/voz/pytorch/test/dynamo/test_distributed.py:19 in foo, code: return flat_param._same_storage_size(x, y)
        eq: "b8[2, 2]" = torch.ops.aten.eq.Scalar(arg0_1, 4);  arg0_1 = None
        return (eq,)
        
```

Reply:
(this is clearly wrong)